diff --git a/src/samples/Samples/0_Introduction/asyncAPI/asyncAPI.cu.hip b/src/samples/Samples/0_Introduction/asyncAPI/asyncAPI.cu.hip
index f1ddbbd..e7d0d6a 100644
--- a/src/samples/Samples/0_Introduction/asyncAPI/asyncAPI.cu.hip
+++ b/src/samples/Samples/0_Introduction/asyncAPI/asyncAPI.cu.hip
@@ -142,10 +142,5 @@ int main(int argc, char *argv[]) {
 
   exit(bFinalResults ? EXIT_SUCCESS : EXIT_FAILURE);
 }
-ree(a));
-  checkCudaErrors(hipFree(d_a));
-
-  exit(bFinalResults ? EXIT_SUCCESS : EXIT_FAILURE);
-}
 
 
