diff --git a/src/samples/Samples/0_Introduction/UnifiedMemoryStreams/UnifiedMemoryStreams.cu.hip b/src/samples/Samples/0_Introduction/UnifiedMemoryStreams/UnifiedMemoryStreams.cu.hip
index 8c29910..bdd07e7 100644
--- a/src/samples/Samples/0_Introduction/UnifiedMemoryStreams/UnifiedMemoryStreams.cu.hip
+++ b/src/samples/Samples/0_Introduction/UnifiedMemoryStreams/UnifiedMemoryStreams.cu.hip
@@ -341,15 +341,3 @@ int main(int argc, char **argv) {
   printf("All Done!\n");
   exit(EXIT_SUCCESS);
 }
-cuBlas handles
-  for (int i = 0; i < nthreads + 1; i++) {
-    hipStreamDestroy(streams[i]);
-    hipblasDestroy(handles[i]);
-  }
-
-  // Free TaskList
-  std::vector<Task<double> >().swap(TaskList);
-
-  printf("All Done!\n");
-  exit(EXIT_SUCCESS);
-}
diff --git a/src/samples/Samples/0_Introduction/asyncAPI/asyncAPI.cu.hip.rej b/src/samples/Samples/0_Introduction/asyncAPI/asyncAPI.cu.hip.rej
deleted file mode 100644
index 88e3083..0000000
--- a/src/samples/Samples/0_Introduction/asyncAPI/asyncAPI.cu.hip.rej
+++ /dev/null
@@ -1,12 +0,0 @@
-diff a/src/samples/Samples/0_Introduction/asyncAPI/asyncAPI.cu.hip b/src/samples/Samples/0_Introduction/asyncAPI/asyncAPI.cu.hip	(rejected hunks)
-@@ -142,10 +142,5 @@ int main(int argc, char *argv[]) {
- 
-   exit(bFinalResults ? EXIT_SUCCESS : EXIT_FAILURE);
- }
--ree(a));
--  checkCudaErrors(hipFree(d_a));
--
--  exit(bFinalResults ? EXIT_SUCCESS : EXIT_FAILURE);
--}
- 
- 
diff --git a/src/samples/Samples/0_Introduction/c++11_cuda/c++11_cuda.cu.hip b/src/samples/Samples/0_Introduction/c++11_cuda/c++11_cuda.cu.hip
index 07994ff..c9c08f2 100644
--- a/src/samples/Samples/0_Introduction/c++11_cuda/c++11_cuda.cu.hip
+++ b/src/samples/Samples/0_Introduction/c++11_cuda/c++11_cuda.cu.hip
@@ -141,7 +141,3 @@ int main(int argc, char **argv) {
 
   return EXIT_SUCCESS;
 }
-rors(hipFree(d_text));
-
-  return EXIT_SUCCESS;
-}
diff --git a/src/samples/Samples/0_Introduction/clock/clock.cu.hip b/src/samples/Samples/0_Introduction/clock/clock.cu.hip
index 9abf99a..53922c3 100644
--- a/src/samples/Samples/0_Introduction/clock/clock.cu.hip
+++ b/src/samples/Samples/0_Introduction/clock/clock.cu.hip
@@ -154,7 +154,3 @@ int main(int argc, char **argv) {
 
   return EXIT_SUCCESS;
 }
- = %Lf\n", avgElapsedClocks);
-
-  return EXIT_SUCCESS;
-}
diff --git a/src/samples/Samples/0_Introduction/concurrentKernels/concurrentKernels.cu.hip b/src/samples/Samples/0_Introduction/concurrentKernels/concurrentKernels.cu.hip
index 895086b..5431f8b 100644
--- a/src/samples/Samples/0_Introduction/concurrentKernels/concurrentKernels.cu.hip
+++ b/src/samples/Samples/0_Introduction/concurrentKernels/concurrentKernels.cu.hip
@@ -229,10 +229,3 @@ int main(int argc, char **argv) {
   printf("Test passed\n");
   exit(EXIT_SUCCESS);
 }
-Test failed!\n");
-    exit(EXIT_FAILURE);
-  }
-
-  printf("Test passed\n");
-  exit(EXIT_SUCCESS);
-}
diff --git a/src/samples/Samples/0_Introduction/cppIntegration/cppIntegration.cu.hip b/src/samples/Samples/0_Introduction/cppIntegration/cppIntegration.cu.hip
index 3b6a637..33b4136 100644
--- a/src/samples/Samples/0_Introduction/cppIntegration/cppIntegration.cu.hip
+++ b/src/samples/Samples/0_Introduction/cppIntegration/cppIntegration.cu.hip
@@ -173,8 +173,3 @@ extern "C" bool runTest(const int argc, const char **argv, char *data,
 
   return success;
 }
-ee(reference);
-  free(reference2);
-
-  return success;
-}
diff --git a/src/samples/Samples/0_Introduction/cppOverload/cppOverload.cu.hip b/src/samples/Samples/0_Introduction/cppOverload/cppOverload.cu.hip
index 78dac7d..00bbb9c 100644
--- a/src/samples/Samples/0_Introduction/cppOverload/cppOverload.cu.hip
+++ b/src/samples/Samples/0_Introduction/cppOverload/cppOverload.cu.hip
@@ -188,10 +188,3 @@ int main(int argc, const char *argv[]) {
 
   exit(testResult ? EXIT_SUCCESS : EXIT_FAILURE);
 }
-tFree(hOutput));
-  checkCudaErrors(hipHostFree(hInput));
-
-  checkCudaErrors(hipDeviceSynchronize());
-
-  exit(testResult ? EXIT_SUCCESS : EXIT_FAILURE);
-}
diff --git a/src/samples/Samples/0_Introduction/cudaOpenMP/cudaOpenMP.cu.hip b/src/samples/Samples/0_Introduction/cudaOpenMP/cudaOpenMP.cu.hip
index a826d91..ecc2755 100644
--- a/src/samples/Samples/0_Introduction/cudaOpenMP/cudaOpenMP.cu.hip
+++ b/src/samples/Samples/0_Introduction/cudaOpenMP/cudaOpenMP.cu.hip
@@ -160,5 +160,3 @@ int main(int argc, char *argv[]) {
 
   exit(bResult ? EXIT_SUCCESS : EXIT_FAILURE);
 }
-  exit(bResult ? EXIT_SUCCESS : EXIT_FAILURE);
-}
diff --git a/src/samples/Samples/0_Introduction/fp16ScalarProduct/fp16ScalarProduct.cu.hip b/src/samples/Samples/0_Introduction/fp16ScalarProduct/fp16ScalarProduct.cu.hip
index 6af6b49..c7203fd 100644
--- a/src/samples/Samples/0_Introduction/fp16ScalarProduct/fp16ScalarProduct.cu.hip
+++ b/src/samples/Samples/0_Introduction/fp16ScalarProduct/fp16ScalarProduct.cu.hip
@@ -215,9 +215,4 @@ int main(int argc, char *argv[]) {
   HIPCHECK(hipHostFree(results));
 
   return EXIT_SUCCESS;
-}
-ee(devResults));
-  checkCudaErrors(hipHostFree(results));
-
-  return EXIT_SUCCESS;
-}
+}
\ No newline at end of file
diff --git a/src/samples/Samples/0_Introduction/matrixMul/matrixMul.cu.hip b/src/samples/Samples/0_Introduction/matrixMul/matrixMul.cu.hip
index 63819e1..2486f8f 100644
--- a/src/samples/Samples/0_Introduction/matrixMul/matrixMul.cu.hip
+++ b/src/samples/Samples/0_Introduction/matrixMul/matrixMul.cu.hip
@@ -354,11 +354,3 @@ int main(int argc, char **argv) {
 
   exit(matrix_result);
 }
-sB.x, dimsB.y);
-
-  checkCudaErrors(rocprofiler_group_start());
-  int matrix_result = MatrixMultiply(argc, argv, block_size, dimsA, dimsB);
-  checkCudaErrors(rocprofiler_group_stop());
-
-  exit(matrix_result);
-}
diff --git a/src/samples/Samples/0_Introduction/mergeSort/mergeSort.cu.hip b/src/samples/Samples/0_Introduction/mergeSort/mergeSort.cu.hip
index 304def2..b3fe541 100644
--- a/src/samples/Samples/0_Introduction/mergeSort/mergeSort.cu.hip
+++ b/src/samples/Samples/0_Introduction/mergeSort/mergeSort.cu.hip
@@ -530,10 +530,4 @@ extern "C" void mergeSort(uint *d_DstKey, uint *d_DstVal, uint *d_BufKey,
     oval = t;
   }
 }
-key;
-    okey = t;
-    t = ival;
-    ival = oval;
-    oval = t;
-  }
-}
+
diff --git a/src/samples/Samples/0_Introduction/simpleAWBarrier/simpleAWBarrier.cu.hip b/src/samples/Samples/0_Introduction/simpleAWBarrier/simpleAWBarrier.cu.hip
index 70e9a70..4ff5608 100644
--- a/src/samples/Samples/0_Introduction/simpleAWBarrier/simpleAWBarrier.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleAWBarrier/simpleAWBarrier.cu.hip
@@ -256,9 +256,3 @@ int runNormVecByDotProductAWBarrier(int argc, char **argv, int deviceId) {
   HIPCHECK(hipHostFree(vecB));
   return matches == size;
 }
-eckCudaErrors(hipFree(d_partialResults));
-
-  checkCudaErrors(hipHostFree(vecA));
-  checkCudaErrors(hipHostFree(vecB));
-  return matches == size;
-}
diff --git a/src/samples/Samples/0_Introduction/simpleAtomicIntrinsics/simpleAtomicIntrinsics.cu.hip b/src/samples/Samples/0_Introduction/simpleAtomicIntrinsics/simpleAtomicIntrinsics.cu.hip
index 88b84f4..12659b3 100644
--- a/src/samples/Samples/0_Introduction/simpleAtomicIntrinsics/simpleAtomicIntrinsics.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleAtomicIntrinsics/simpleAtomicIntrinsics.cu.hip
@@ -134,6 +134,3 @@ void runTest(int argc, char **argv) {
   HIPCHECK(hipHostFree(hOData));
   HIPCHECK(hipFree(dOData));
 }
-ostFree(hOData));
-  checkCudaErrors(hipFree(dOData));
-}
diff --git a/src/samples/Samples/0_Introduction/simpleAttributes/simpleAttributes.cu.hip b/src/samples/Samples/0_Introduction/simpleAttributes/simpleAttributes.cu.hip
index 048cca4..d2ef5cc 100644
--- a/src/samples/Samples/0_Introduction/simpleAttributes/simpleAttributes.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleAttributes/simpleAttributes.cu.hip
@@ -214,9 +214,4 @@ void runTest(int argc, char **argv) {
   sdkDeleteTimer(&timer);
 
   exit(bTestResult ? EXIT_SUCCESS : EXIT_FAILURE);
-}
-%f (ms)\n", sdkGetTimerValue(&timer));
-  sdkDeleteTimer(&timer);
-
-  exit(bTestResult ? EXIT_SUCCESS : EXIT_FAILURE);
-}
+}
\ No newline at end of file
diff --git a/src/samples/Samples/0_Introduction/simpleCallback/simpleCallback.cu.hip b/src/samples/Samples/0_Introduction/simpleCallback/simpleCallback.cu.hip
index 3585331..97459db 100644
--- a/src/samples/Samples/0_Introduction/simpleCallback/simpleCallback.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleCallback/simpleCallback.cu.hip
@@ -215,9 +215,3 @@ int main(int argc, char **argv) {
 
   exit(success ? EXIT_SUCCESS : EXIT_FAILURE);
 }
-ccess" : "Failure");
-
-  free(workloads);
-
-  exit(success ? EXIT_SUCCESS : EXIT_FAILURE);
-}
diff --git a/src/samples/Samples/0_Introduction/simpleCubemapTexture/simpleCubemapTexture.cu.hip b/src/samples/Samples/0_Introduction/simpleCubemapTexture/simpleCubemapTexture.cu.hip
index 8a9eb14..b7537d0 100644
--- a/src/samples/Samples/0_Introduction/simpleCubemapTexture/simpleCubemapTexture.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleCubemapTexture/simpleCubemapTexture.cu.hip
@@ -269,7 +269,3 @@ int main(int argc, char **argv) {
 
   exit(bResult ? EXIT_SUCCESS : EXIT_FAILURE);
 }
-Errors(hipFreeArray(cu_3darray));
-
-  exit(bResult ? EXIT_SUCCESS : EXIT_FAILURE);
-}
diff --git a/src/samples/Samples/0_Introduction/simpleHyperQ/simpleHyperQ.cu.hip b/src/samples/Samples/0_Introduction/simpleHyperQ/simpleHyperQ.cu.hip
index 748edb6..59cfcfc 100644
--- a/src/samples/Samples/0_Introduction/simpleHyperQ/simpleHyperQ.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleHyperQ/simpleHyperQ.cu.hip
@@ -232,8 +232,3 @@ int main(int argc, char **argv) {
 
   exit(bTestResult ? EXIT_SUCCESS : EXIT_FAILURE);
 }
-pHostFree(a);
-  hipFree(d_a);
-
-  exit(bTestResult ? EXIT_SUCCESS : EXIT_FAILURE);
-}
diff --git a/src/samples/Samples/0_Introduction/simpleIPC/simpleIPC.cu.hip b/src/samples/Samples/0_Introduction/simpleIPC/simpleIPC.cu.hip
index bdef607..050f9cc 100644
--- a/src/samples/Samples/0_Introduction/simpleIPC/simpleIPC.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleIPC/simpleIPC.cu.hip
@@ -330,15 +330,3 @@ int main(int argc, char **argv) {
   return EXIT_SUCCESS;
 #endif
 }
-__arm__) || defined(__aarch64__)
-  printf("Not supported on ARM\n");
-  return EXIT_WAIVED;
-#else
-  if (argc == 1) {
-    parentProcess(argv[0]);
-  } else {
-    childProcess(atoi(argv[1]));
-  }
-  return EXIT_SUCCESS;
-#endif
-}
diff --git a/src/samples/Samples/0_Introduction/simpleLayeredTexture/simpleLayeredTexture.cu.hip b/src/samples/Samples/0_Introduction/simpleLayeredTexture/simpleLayeredTexture.cu.hip
index c49b407..80bf3fc 100644
--- a/src/samples/Samples/0_Introduction/simpleLayeredTexture/simpleLayeredTexture.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleLayeredTexture/simpleLayeredTexture.cu.hip
@@ -217,7 +217,3 @@ int main(int argc, char **argv) {
 
   exit(bResult ? EXIT_SUCCESS : EXIT_FAILURE);
 }
-hipFreeArray(cu_3darray));
-
-  exit(bResult ? EXIT_SUCCESS : EXIT_FAILURE);
-}
diff --git a/src/samples/Samples/0_Introduction/simpleMultiCopy/simpleMultiCopy.cu.hip b/src/samples/Samples/0_Introduction/simpleMultiCopy/simpleMultiCopy.cu.hip
index faead55..446b8a8 100644
--- a/src/samples/Samples/0_Introduction/simpleMultiCopy/simpleMultiCopy.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleMultiCopy/simpleMultiCopy.cu.hip
@@ -373,11 +373,4 @@ bool test() {
   }
 
   return passed;
-}
-    for (int i = 0; i < N; ++i) {
-      passed &= (h_data_out[j][i] == 1);
-    }
-  }
-
-  return passed;
-}
+}
\ No newline at end of file
diff --git a/src/samples/Samples/0_Introduction/simpleMultiGPU/simpleMultiGPU.cu.hip b/src/samples/Samples/0_Introduction/simpleMultiGPU/simpleMultiGPU.cu.hip
index 219be98..b4fdd44 100644
--- a/src/samples/Samples/0_Introduction/simpleMultiGPU/simpleMultiGPU.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleMultiGPU/simpleMultiGPU.cu.hip
@@ -237,9 +237,3 @@ int main(int argc, char **argv) {
 
   exit((diff < 1e-5) ? EXIT_SUCCESS : EXIT_FAILURE);
 }
-ice(i));
-    checkCudaErrors(hipHostFree(plan[i].h_Data));
-  }
-
-  exit((diff < 1e-5) ? EXIT_SUCCESS : EXIT_FAILURE);
-}
diff --git a/src/samples/Samples/0_Introduction/simpleOccupancy/simpleOccupancy.cu.hip b/src/samples/Samples/0_Introduction/simpleOccupancy/simpleOccupancy.cu.hip
index fbc73d8..dd701c1 100644
--- a/src/samples/Samples/0_Introduction/simpleOccupancy/simpleOccupancy.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleOccupancy/simpleOccupancy.cu.hip
@@ -241,11 +241,3 @@ int main() {
 
   return 0;
 }
-rn -1;
-  }
-
-  std::cout << std::endl;
-  std::cout << "Test PASSED\n" << std::endl;
-
-  return 0;
-}
diff --git a/src/samples/Samples/0_Introduction/simpleP2P/simpleP2P.cu.hip b/src/samples/Samples/0_Introduction/simpleP2P/simpleP2P.cu.hip
index 5300ee9..9ef2e42 100644
--- a/src/samples/Samples/0_Introduction/simpleP2P/simpleP2P.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleP2P/simpleP2P.cu.hip
@@ -261,18 +261,3 @@ int main(int argc, char **argv) {
     exit(EXIT_SUCCESS);
   }
 }
-(g1));
-  checkCudaErrors(hipHostFree(h0));
-
-  for (int i = 0; i < gpu_n; i++) {
-    checkCudaErrors(hipSetDevice(i));
-  }
-
-  if (error_count != 0) {
-    printf("Test failed!\n");
-    exit(EXIT_FAILURE);
-  } else {
-    printf("Test passed\n");
-    exit(EXIT_SUCCESS);
-  }
-}
diff --git a/src/samples/Samples/0_Introduction/simplePitchLinearTexture/simplePitchLinearTexture.cu.hip b/src/samples/Samples/0_Introduction/simplePitchLinearTexture/simplePitchLinearTexture.cu.hip
index c1d9525..92ba449 100644
--- a/src/samples/Samples/0_Introduction/simplePitchLinearTexture/simplePitchLinearTexture.cu.hip
+++ b/src/samples/Samples/0_Introduction/simplePitchLinearTexture/simplePitchLinearTexture.cu.hip
@@ -312,10 +312,3 @@ void runTest(int argc, char **argv) {
   HIPCHECK(hipEventDestroy(start));
   HIPCHECK(hipEventDestroy(stop));
 }
-dataPL));
-  checkCudaErrors(hipFreeArray(d_idataArray));
-  checkCudaErrors(hipFree(d_odata));
-
-  checkCudaErrors(hipEventDestroy(start));
-  checkCudaErrors(hipEventDestroy(stop));
-}
diff --git a/src/samples/Samples/0_Introduction/simplePrintf/simplePrintf.cu.hip b/src/samples/Samples/0_Introduction/simplePrintf/simplePrintf.cu.hip
index 19d32d7..519e66c 100644
--- a/src/samples/Samples/0_Introduction/simplePrintf/simplePrintf.cu.hip
+++ b/src/samples/Samples/0_Introduction/simplePrintf/simplePrintf.cu.hip
@@ -74,5 +74,3 @@ int main(int argc, char **argv) {
 
   return EXIT_SUCCESS;
 }
-IT_SUCCESS;
-}
diff --git a/src/samples/Samples/0_Introduction/simpleStreams/simpleStreams.cu.hip b/src/samples/Samples/0_Introduction/simpleStreams/simpleStreams.cu.hip
index 6722117..14cf24c 100644
--- a/src/samples/Samples/0_Introduction/simpleStreams/simpleStreams.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleStreams/simpleStreams.cu.hip
@@ -446,13 +446,3 @@ int main(int argc, char **argv) {
 
   return bResults ? EXIT_SUCCESS : EXIT_FAILURE;
 }
-y(stop_event));
-
-  // Free hipHostMalloc or Generic Host allocated memory (from CUDA 4.0)
-  FreeHostMemory(bPinGenericMemory, &h_a, &hAligned_a, nbytes);
-
-  checkCudaErrors(hipFree(d_a));
-  checkCudaErrors(hipFree(d_c));
-
-  return bResults ? EXIT_SUCCESS : EXIT_FAILURE;
-}
diff --git a/src/samples/Samples/0_Introduction/simpleSurfaceWrite/simpleSurfaceWrite.cu.hip b/src/samples/Samples/0_Introduction/simpleSurfaceWrite/simpleSurfaceWrite.cu.hip
index 8b7d3d7..11b96bb 100644
--- a/src/samples/Samples/0_Introduction/simpleSurfaceWrite/simpleSurfaceWrite.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleSurfaceWrite/simpleSurfaceWrite.cu.hip
@@ -296,8 +296,3 @@ void runTest(int argc, char **argv) {
   free(imagePath);
   free(refPath);
 }
-e(dData));
-  checkCudaErrors(hipFreeArray(cuArray));
-  free(imagePath);
-  free(refPath);
-}
diff --git a/src/samples/Samples/0_Introduction/simpleTemplates/simpleTemplates.cu.hip b/src/samples/Samples/0_Introduction/simpleTemplates/simpleTemplates.cu.hip
index 2229723..b3f8428 100644
--- a/src/samples/Samples/0_Introduction/simpleTemplates/simpleTemplates.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleTemplates/simpleTemplates.cu.hip
@@ -265,6 +265,3 @@ void runTest(int argc, char **argv, int len) {
   HIPCHECK(hipFree(d_idata));
   HIPCHECK(hipFree(d_odata));
 }
-_idata));
-  checkCudaErrors(hipFree(d_odata));
-}
diff --git a/src/samples/Samples/0_Introduction/simpleTexture/simpleTexture.cu.hip b/src/samples/Samples/0_Introduction/simpleTexture/simpleTexture.cu.hip
index 98da336..5024cdd 100644
--- a/src/samples/Samples/0_Introduction/simpleTexture/simpleTexture.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleTexture/simpleTexture.cu.hip
@@ -252,7 +252,3 @@ void runTest(int argc, char **argv) {
   free(imagePath);
   free(refPath);
 }
-aErrors(hipFreeArray(cuArray));
-  free(imagePath);
-  free(refPath);
-}
diff --git a/src/samples/Samples/0_Introduction/simpleTexture3D/simpleTexture3D_kernel.cu.hip b/src/samples/Samples/0_Introduction/simpleTexture3D/simpleTexture3D_kernel.cu.hip
index 2e9dc28..1cfce6b 100644
--- a/src/samples/Samples/0_Introduction/simpleTexture3D/simpleTexture3D_kernel.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleTexture3D/simpleTexture3D_kernel.cu.hip
@@ -141,4 +141,3 @@ void cleanupCuda() {
 #endif  // #ifndef _SIMPLETEXTURE3D_KERNEL_CU_
 
 
-#endif  // #ifndef _SIMPLETEXTURE3D_KERNEL_CU_
diff --git a/src/samples/Samples/0_Introduction/simpleVoteIntrinsics/simpleVoteIntrinsics.cu.hip b/src/samples/Samples/0_Introduction/simpleVoteIntrinsics/simpleVoteIntrinsics.cu.hip
index e3f6d71..ad5e474 100644
--- a/src/samples/Samples/0_Introduction/simpleVoteIntrinsics/simpleVoteIntrinsics.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleVoteIntrinsics/simpleVoteIntrinsics.cu.hip
@@ -309,7 +309,3 @@ int main(int argc, char **argv) {
              ? EXIT_SUCCESS
              : EXIT_FAILURE;
 }
-_count[1] == 0 && error_count[2] == 0)
-             ? EXIT_SUCCESS
-             : EXIT_FAILURE;
-}
diff --git a/src/samples/Samples/0_Introduction/simpleZeroCopy/simpleZeroCopy.cu.hip b/src/samples/Samples/0_Introduction/simpleZeroCopy/simpleZeroCopy.cu.hip
index adf68a1..42fa66a 100644
--- a/src/samples/Samples/0_Introduction/simpleZeroCopy/simpleZeroCopy.cu.hip
+++ b/src/samples/Samples/0_Introduction/simpleZeroCopy/simpleZeroCopy.cu.hip
@@ -247,11 +247,4 @@ int main(int argc, char **argv) {
   }
 
   exit(errorNorm / refNorm < 1.e-6f ? EXIT_SUCCESS : EXIT_FAILURE);
-}
-HostFree(b));
-    checkCudaErrors(hipHostFree(c));
-#endif
-  }
-
-  exit(errorNorm / refNorm < 1.e-6f ? EXIT_SUCCESS : EXIT_FAILURE);
-}
+}
\ No newline at end of file
diff --git a/src/samples/Samples/0_Introduction/systemWideAtomics/systemWideAtomics.cu.hip b/src/samples/Samples/0_Introduction/systemWideAtomics/systemWideAtomics.cu.hip
index 268c31b..7908c82 100644
--- a/src/samples/Samples/0_Introduction/systemWideAtomics/systemWideAtomics.cu.hip
+++ b/src/samples/Samples/0_Introduction/systemWideAtomics/systemWideAtomics.cu.hip
@@ -343,5 +343,3 @@ int main(int argc, char **argv) {
          testResult ? "OK" : "ERROR!");
   exit(testResult ? EXIT_SUCCESS : EXIT_FAILURE);
 }
-S : EXIT_FAILURE);
-}
